src/components/Dashboard/ThisWeek/NewsletterMenu/MenuItems/ExportPhysical/ExportPhysical.test.tsx

```diff
@@ -41,6 +52,121 @@ describe('ExportPhysical', () => {
         />
       </MockedProvider>,
     );
-    expect(queryByText('Export Contacts')).toBeInTheDocument();
+    expect(getByText('Export Contacts')).toBeInTheDocument();
+  });
+
+  describe('Exporting Contacts', () => {
+    const { location } = window;
+
+    beforeAll(() => {
+      // Have to ignore TS complaining about deleting window.location because location is not optional
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      delete window.location;
+      window.location = { ...location, replace: jest.fn() };
+    });
+
+    afterAll(() => {
+      window.location = location;
+    });
+
+    it('Exports Contacts and Downloads File', async () => {
+      const url = 'someRandomUrlToFile';
+      const mocks = {
+        CreateExportedContacts: {
+          exportContacts: url,
+        },
+      };
+      const { getByText } = render(
+        <GqlMockedProvider<CreateExportedContactsMutation> mocks={mocks}>
+          <ExportPhysical
+            accountListId={accountListId}
+            handleClose={handleClose}
+          />
+        </GqlMockedProvider>,
+      );
+      userEvent.click(getByText('Advanced CSV'));
+      await waitFor(() =>
+        expect(window.location.replace).toHaveBeenCalledWith(
+          `${url}?access_token=someToken1234`,
+        ),
+      );
+    });
+  });
+
+  describe('Exporting Contacts Mutation', () => {
+    // it('Exports Contacts - PDF of Mail Merged Labels', async () => {
```

Comment:
@OzzieOrca I was talking with @gyasistory and he mentioned a way of testing that you suggested, [example here](https://github.com/CruGlobal/mpdx-react/blob/d3f052ff6aa0007fe0bb97cb513da110b49456b7/src/components/Contacts/ContactDetails/ContactDetailsTab/ContactDetailsTab.test.tsx#L7). So I tried to implement this for the mutation. However, I'm getting a ```timed out in waitfornextupdate after 1000ms``` error from the ```waitForNextUpdate```. Any ideas why? I tried looking around but the suggestions I found didn't seem to work. Also, do we want to even do this way of testing with mutations? Would love to get your opinion on which test format would be better. Or maybe even having both would be fine 🤷‍♂️

Reply:
Umm I don't think we need to test generated code. That should be the responsibility of the Apollo library and codegen libraries. We should test by interacting with our component as a user would by clicking a button or something.

I'm trying to figure out the context of what I suggested...

It's probably not working cuz it's a mutation and you never called the mutation function before calling `await waitForNextUpdate();` so there were no changes.